pkg/await/apps_deployment.go

```diff
@@ -455,8 +455,32 @@ func (dia *deploymentInitAwaiter) checkReplicaSetStatus() {
 	glog.V(3).Infof("ReplicaSet '%s' requests '%v' replicas, but has '%v' ready",
 		rs.GetName(), specReplicas, readyReplicas)
 
-	dia.updatedReplicaSetReady = lastGeneration != dia.currentGeneration && udpatedReplicaSetCreated &&
-		readyReplicasExists && readyReplicas >= int64(specReplicas)
+	if dia.changeTriggeredRollout() {
+		dia.updatedReplicaSetReady = lastGeneration != dia.currentGeneration && udpatedReplicaSetCreated &&
+			readyReplicasExists && readyReplicas >= int64(specReplicas)
+	} else {
+		dia.updatedReplicaSetReady = lastGeneration == dia.currentGeneration && udpatedReplicaSetCreated &&
```

Comment:
Do we need to enforce that they are equal on this path?  In general, how should we think about being conservative about blocking success?  Seems we should aim to have as few things be capable of blocking success as possible, and be sure they are rock solid signals of a problem if they do?

Reply:
Ok, we can do that.